src/mem/largealloc.h

```diff
@@ -202,35 +216,76 @@ namespace snmalloc
     }
 
     template<bool committed>
-    void* reserve(size_t* size, size_t align) noexcept
+    void* reserve(size_t large_class) noexcept
     {
+      size_t size = bits::one_at_bit(SUPERSLAB_BITS) << large_class;
+      size_t align = size;
+
       if constexpr (pal_supports<AlignedAllocation, PAL>)
       {
         return PAL::template reserve<committed>(size, align);
       }
       else
       {
-        size_t request = *size;
-        // Add align, so we can guarantee to provide at least size.
-        request += align;
-        // Alignment must be a power of 2.
-        assert(align == bits::next_pow2(align));
+        // Reserve 4 times the amount, and put aligned leftovers into the
+        // large_stack
+        size_t request = bits::max(size * 4, SUPERSLAB_SIZE * 8);
+        void* p = PAL::template reserve<committed>(request);
+
+        address_t p0 = address_cast(p);
+        address_t start = bits::align_up(p0, align);
+        address_t p1 = p0 + request;
+        address_t end = start + size;
+
+        // Turn off pages for extra allocations.
+        // Will turn pages back on for stack entries.
+        if (committed)

```

Comment:
Wouldn't it be better to `reserve<false>` above and then commit the required bits here, rather than committing everything above and then decommitting here?  The current flow requires that the OS commit a big chunk and then returns most of it almost immediately.